docs/configuration/settings.rst

```diff
@@ -569,6 +569,172 @@ You can set ``account_create_principals`` if you want to limit account creation
 
 See the :ref:`API docs <api-accounts>` to create accounts, change passwords etc.
 
+.. _settings-account-validation:
+
+**About account validation**
+
+You can enable the :ref:`account validation <accounts-validate>` option, which
+will require account IDs to be valid email addresses, to which a validation
+email will be sent with an activation key.
+
+.. code-block:: ini
+
+    kinto.account_validation = true
+    # Mail configuration:
+    # Set the sender for the validation email.
+    kinto.account_validation.email_sender = "admin@example.com"
+
+.. note::
+
+    Both the account validation and password reset need a properly configured
+    smtp server.
+    To use a debug or testing mailer you may use the ``mail.mailer = debug`` or
+    ``mail.mailer = testing`` settings. Refer to
+    `pyramid_mailer's configuration <https://docs.pylonsproject.org/projects/pyramid_mailer/en/latest/#configuration>`_.
+
+You can restrict the email addresses allowed using the
+``account_validation.email_regexp`` setting, and the delay for which the
+activation key will be valid:
+
+.. code-block:: ini
+
+    # Set the regular expression used to validate a proper email address.
+    kinto.account_validation.email_regexp = "^[a-zA-Z0-9_.+-]+@[a-zA-Z0-9-]+\\.[a-zA-Z0-9-.]+$"
```

Comment:
Here you accept `+` character

Reply:
that's an "additional restriction", it doesn't take over/replace what's declared in https://github.com/Kinto/kinto/blob/master/kinto/plugins/accounts/views.py#L33 (if that's the root cause of this issue, not sure yet, I'm investigating)

Reply:
Maybe shall we use the same regex for both?

Reply:
I don't think it's the same use case: in one case (no account validation) you want to allow a few set of characters, in the other case you might want to restrict emails to a given sub-domain.

Reply:
I pushed a fix

Reply:
> I don't think it's the same use case: in one case (no account validation) you want to allow a few set of characters, in the other case you might want to restrict emails to a given sub-domain.

Yeah makes sense :+1: